src/main/webapp/app/code-editor/service/code-editor-repository.service.ts

```diff
@@ -122,16 +153,39 @@ export class CodeEditorRepositoryFileService extends DomainDependentEndpoint imp
         this.fileUpdateSubject = new Subject<{ [p: string]: string | null }>();
 
         this.jhiWebsocketService.subscribe(this.fileUpdateUrl);
-        this.jhiWebsocketService.receive(this.fileUpdateUrl).subscribe(res => this.fileUpdateSubject.next(res), err => this.fileUpdateSubject.error(err));
+        this.jhiWebsocketService
+            .receive(this.fileUpdateUrl)
+            .pipe(this.handleWebsocketErrorResponse())
+            .subscribe(res => this.fileUpdateSubject.next(res), err => this.fileUpdateSubject.error(err));
         this.jhiWebsocketService.send(`${this.websocketResourceUrlSend}/files`, fileUpdates);
         return this.fileUpdateSubject as Observable<{ [fileName: string]: string | null }>;
     };
 
     renameFile = (currentFilePath: string, newFilename: string) => {
-        return this.http.post<void>(`${this.restResourceUrl}/rename-file`, { currentFilePath, newFilename });
+        return this.http.post<void>(`${this.restResourceUrl}/rename-file`, { currentFilePath, newFilename }).pipe(this.handleErrorResponse());
     };
 
     deleteFile = (fileName: string) => {
-        return this.http.delete<void>(`${this.restResourceUrl}/file`, { params: new HttpParams().set('file', fileName) });
+        return this.http.delete<void>(`${this.restResourceUrl}/file`, { params: new HttpParams().set('file', fileName) }).pipe(this.handleErrorResponse());
     };
+
+    private handleErrorResponse = <T>(): UnaryFunction<Observable<T>, Observable<T>> =>
+        pipe(
+            catchError((err: HttpErrorResponse) => {
+                if (err.status === 409) {
+                    this.conflictService.notifyConflictState(GitConflictState.CHECKOUT_CONFLICT);
+                }
+                return throwError(err);
+            }),
+        );
+
+    private handleWebsocketErrorResponse = <T>(): UnaryFunction<Observable<T>, Observable<T>> =>
+        pipe(
+            catchError((err: { error: string }) => {
+                if (err.error === 'checkoutConflict') {
```

Comment:
you might want to save 'checkoutConflict' in a constant

Reply:
Good catch!